sdk/keyvault/Azure.Security.KeyVault.Keys/Readme.md

```diff
@@ -0,0 +1,282 @@
+    az ad sp create-for-rbac -n <your-application-name> --skip-assignment
+    ```
+    Output:
+    ```json
+    {
+        "appId": "generated-app-ID",
+        "displayName": "dummy-app-name",
+        "name": "http://dummy-app-name",
+        "password": "random-password",
+        "tenant": "tenant-ID"
+    }
+    ```
+* Use the above returned credentials information to set **AZURE_CLIENT_ID**(appId), **AZURE_CLIENT_SECRET**(password) and **AZURE_TENANT_ID**(tenant) environment variables. The following example shows a way to do this in Powershell:
+```cmd
+$Env:AZURE_CLIENT_ID="generated-app-ID"
+$Env:AZURE_CLIENT_SECRET="random-password"
+$Env:AZURE_TENANT_ID="tenant-ID"
+```
+
+* Grant the above mentioned application authorization to perform key operations on the key vault:
+    ```PowerShell
+    az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --key-permissions backup delete get list set
+    ```
+    > --key-permissions:
+    > Accepted values: backup, delete, get, list, purge, recover, restore, set
+
+* Use the above mentioned Key Vault name to retrieve details of your Vault which also contains your Key Vault URL:
+    ```PowerShell
+    az keyvault show --name <your-key-vault-name> 
+    ```
+
+#### Create Key client
+Once you've populated the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables and replaced **your-vault-url** with the above returned URI, you can create the [KeyClient][key_client_class]:
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Keys;
+
+// Create a new key client using the default credential from Azure.Identity
+var client = new KeyClient(vaultUri: <your-vault-url>, credential: new DefaultAzureCredential());
+
+// Create a new key using the key client
+Key key = await Client.CreateKey("key-name", KeyType.EllipticCurve);
+```
+> DefaultAzureCredential():
+> Uses the environment variables previously set (`AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, and `AZURE_TENANT_ID`).
+
+## Key concepts
+### Keys
+Azure Key Vault supports multiple key types and algorithms, and enables the use of Hardware Security Modules (HSM) for high value keys.
+
+### Key Client:
+An asynchronous and synchronous KeyClient client exists in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a KeyClient, you can interact with the primary resource types in Key Vault.
+
+## Examples
+The Azure.Security.KeyVault.Keys package supports synchronous and asynchronous APIs.
+
+The following section provides several code snippets using the [above created](#create-key-client) `client`, covering some of the most common Azure Key Vault Key service related tasks:
+
+### Sync examples
+* [Create a Key](#create-a-key)
+* [Retrieve a Key](#retrieve-a-key)
+* [Update an existing Key](#update-an-existing-key)
+* [Delete a Key](#delete-a-key)
+* [List Keys](#list-keys)
+
+### Async examples
+* [Create a Key](#async-create-a-key)
```

Comment:
Is this the balance we want between sync/async in the README?  For Storage I was planning on flipping this with one sync example and a handful of async examples.  I can adjust to match this if the @kaerm thinks it's more helpful.

Reply:
That's how we've been doing it in other readmes for KV, and what I understood from Kate. Happy to change in case it is wrong

Reply:
In general I am all for equal representation of sync and async but I think different languages took different approaches to that. From what I understand the main focus was on providing sync libraries first as they are easier to understand for people getting started and then async because people who know they need async will choose it anyway. In general, I would provide all examples for both of them (with sync listed first but clear mention that async exists) but if the samples are very similar and just need an additional word then I am fine with the way it is listed here. Anyway, that's what I would suggest and we can verify if it gets confusing for people to use